src/Agent.Worker/TaskCommandExtension.cs

```diff
@@ -545,6 +545,18 @@ public void Execute(IExecutionContext context, Command command)
                 Boolean.TryParse(isOutputValue, out isOutput);
             }
 
+            String isReadOnlyValue;
+            Boolean isReadOnly = false;
+            if (eventProperties.TryGetValue(TaskSetVariableEventProperties.IsReadOnly, out isReadOnlyValue))
+            {
+                Boolean.TryParse(isReadOnlyValue, out isReadOnly);
+            }
+
+            if (context.Variables.IsReadOnly(name))
+            {
+                context.Warning($"Overwriting readonly variable '{name}'. This behavior will be disabled in the future. See https://github.com/microsoft/azure-pipelines-yaml/blob/master/design/readonly-variables.md for details.");
```

Comment:
should this string be localized?

Reply:
If you have questions on how to do this just shout.  Basically you want to use StringUtil.Loc( like other places in this file.

Reply:
Done - here and below with different messages